src/nnet3/nnet-cudnn-simple-component.cc

```diff
@@ -638,11 +638,11 @@ void CuDNN3DConvolutionComponent::Write(std::ostream &os, bool binary) const {
   KALDI_ASSERT(filter_dims[1] == input_num_filters_);
   KALDI_ASSERT(filter_dims[0] == num_filters_);
   WriteToken(os, binary, "<FilterXDim>");
-  WriteBasicType(os, binary, filter_dims[2]);
```

Comment:
This change is correct. Thank you. CUDNN's 3D convolution assumes tensors are laid out as "NCDHW" format, with the first letter corresponding to index 0, and the last corresponding to index 4.

I will look through the rest of the code to verify that I correctly follow CUDNN's indexing.